google/cloud/spanner/internal/session_pool.cc

```diff
@@ -89,6 +99,45 @@ void SessionPool::Initialize() {
       (void)CreateSessions(lk, channel, options_.labels(), num_sessions);
     }
   }
+  ScheduleBackgroundWork(std::chrono::seconds(5));
+}
+
+SessionPool::~SessionPool() {
+  // Taking `mu_` here is not strictly required because all references to this
+  // object are via `shared_ptr` and the fact we're in the destructor implies
+  // there are no more references.
+  //
+  // Note that it *is* possible we're racing against the timer lambda in
+  // `ScheduleBackgroundWork`. That does have a `weak_ptr` to `this`, but
+  // the `lock()` call (which must not have yet executed, else we wouldn't be
```

Comment:
"not have yet executed" needs to be more precise methinks.  We could, for example, be executing `lock()` right now.  Or it may have just completed execution and returned null.